android/kotlin/v2/AmpliApp/app/src/main/java/com/amplitude/ampli/Ampli.kt

```diff
@@ -0,0 +1,822 @@
+     * @param optionalString Optional String property description
+     */
+    constructor(
+        optionalArrayNumber: Array<Double>? = null,
+        optionalArrayString: Array<String>? = null,
+        optionalBoolean: Boolean? = null,
+        optionalNumber: Double? = null,
+        optionalString: String? = null
+    ) : this() {
+        this.eventType = "EventWithOptionalProperties"
+        this.eventProperties =
+            mapOf(
+                *(if (optionalArrayNumber != null) arrayOf("optionalArrayNumber" to optionalArrayNumber) else arrayOf()),
+                *(if (optionalArrayString != null) arrayOf("optionalArrayString" to optionalArrayString) else arrayOf()),
+                *(if (optionalBoolean != null) arrayOf("optionalBoolean" to optionalBoolean) else arrayOf()),
+                *(if (optionalNumber != null) arrayOf("optionalNumber" to optionalNumber) else arrayOf()),
+                *(if (optionalString != null) arrayOf("optionalString" to optionalString) else arrayOf())
+            ).toMutableMap()
+    }
+  }
+
+val ampli = Ampli()
+
+open class Ampli {
+    companion object {
+        val API_KEY: Map<Environment, String> = mapOf(
+            Environment.DEVELOPMENT to "",
+            Environment.PRODUCTION to ""
+        )
+
+		private val observePlan: Plan = Plan("main", "kotlin-ampli-v2", "0", "79154a50-f057-4db5-9755-775e4e9f05e6")
+    }
+
+    enum class Environment {
+        DEVELOPMENT,
+        PRODUCTION
+    }
+
+    private var disabled: Boolean = false
+
+    @android.annotation.SuppressLint("StaticFieldLeak")
+    private var _client: Amplitude? = null
+
+    open fun load(appContext: android.content.Context, options: LoadOptions? = null) {
+        this.disabled = options?.disabled ?: false
+        if (this._client != null) {
+            System.err.println("Warning: Ampli is already initialized. ampli.load() should be called once at application start up.")
+            return
+        }
+        val env = options?.environment ?: Environment.DEVELOPMENT
+        val apiKey = options?.client?.apiKey ?: API_KEY[env]
+        val plan = options?.client?.plan ?: observePlan
+        when {
+            options?.client?.instance != null -> {
+                this._client = options.client.instance
+            }
+            apiKey != null && apiKey != "" -> {
+                this._client = Amplitude(
+                    Configuration(
+                        apiKey = apiKey,
+                        plan = plan,
+                        context = appContext.applicationContext
+                    )
+                )
+            }
+            else -> {
+                System.err.println("ampli.load() requires 'environment', 'client.apiKey', or 'client.instance'")
+                return
+            }
+        }
+    }
+
+    open fun track(event: BaseEvent, options: EventOptions? = null) {
+		if (!isInitializedAndEnabled()) {
```

Comment:
indent

Reply:
<img width="605" alt="Screen Shot 2022-05-27 at 10 21 44 PM" src="https://user-images.githubusercontent.com/13028329/170811384-55ab9c44-2d35-4c34-b86d-9ead61aabeb9.png">
The indent looks right in the IDE but not right in the GitHub for some reason.